tests/vmi_multus_test.go

```diff
@@ -580,6 +580,36 @@ var _ = Describe("SRIOV", func() {
 
 	tests.BeforeAll(func() {
 		tests.BeforeTestCleanup()
+		// Check if the hardware supports SRIOV
+		nodes := tests.GetAllSchedulableNodes(virtClient)
+		Expect(nodes.Items).ToNot(BeEmpty(), "There should be some compute node")
+
+		for _, node := range nodes.Items {
+			var sriovallocate bool = false
+			var sriovcapacity bool = false
+			resourceList := node.Status.Allocatable
+			for k, v := range resourceList {
+				if strings.HasPrefix(string(k), sriovResourceName) {
+					if v.Value() > 0 {
+						sriovallocate = true
+						break
+					}
+				}
+			}
+			capacityList := node.Status.Capacity
+			for k, v := range capacityList {
```

Comment:
If you store the whole check in a function, you can perform an early return if you don't find even the resource. Then you can avoid this second iteration.

Reply:
Done